code/datums/chatmessage.dm

```diff
@@ -129,13 +133,24 @@
 	if (!ismob(target))
 		extra_classes |= "small"
 
+	var/list/prefixes
+
 	// Append radio icon if from a virtual speaker
 	if (extra_classes.Find("virtual-speaker"))
 		var/image/r_icon = image('icons/UI_Icons/chat/chat_icons.dmi', icon_state = "radio")
-		text =  "\icon[r_icon]&nbsp;[text]"
+		LAZYADD(prefixes, "\icon[r_icon]")
 	else if (extra_classes.Find("emote"))
 		var/image/r_icon = image('icons/UI_Icons/chat/chat_icons.dmi', icon_state = "emote")
-		text =  "\icon[r_icon]&nbsp;[text]"
+		LAZYADD(prefixes, "\icon[r_icon]")
+
+	// Append language icon if the language uses one
+	var/datum/language/language_instance = GLOB.language_datum_instances[language]
+	if (language_instance?.display_icon(owner))
+		var/icon/language_icon = icon(language_instance.icon, icon_state = language_instance.icon_state)
+		language_icon.Scale(CHAT_MESSAGE_ICON_SIZE, CHAT_MESSAGE_ICON_SIZE)
```

Comment:
I don't know much about BYOND performance, but something about this gives me the chills. Should I be concerned? Should this just be cached, or is there no notable performance gain to be had?

Reply:
@AnturK ? @SpaceManiac ? Someone needs to answer this because I'm concerned about it as well

Reply:
Runechat already has performance issues at higher pop, adding blend operations to it sounds like horrible idea.

Reply:
Can we cache these scalings and send it as an asset?

Reply:
This or using transform at least.